scripts/build-windows.ps1

```diff
@@ -0,0 +1,175 @@
+
+# select the appropriate cmake generator string given the environment
+if ( $VS_VERSION -eq '2019' ) { $CMAKE_GENERATOR = 'Visual Studio 16 2019'; $MSBUILDVER = "16.0"; }
+if ( $VS_VERSION -eq '2015' -and $DEVENV_PLATFORM -eq 'Win32' ) { $CMAKE_GENERATOR = 'Visual Studio 14 2015'; $MSBUILDVER = "14.0"; }
+if ( $VS_VERSION -eq '2015' -and $DEVENV_PLATFORM -eq 'x64' ) { $CMAKE_GENERATOR = 'Visual Studio 14 2015 Win64'; $MSBUILDVER = "14.0"; }
+if ( $VS_VERSION -eq '2013' -and $DEVENV_PLATFORM -eq 'Win32' ) { $CMAKE_GENERATOR = 'Visual Studio 12 2013'; $MSBUILDVER = "12.0"; }
+if ( $VS_VERSION -eq '2013' -and $DEVENV_PLATFORM -eq 'x64' ) { $CMAKE_GENERATOR = 'Visual Studio 12 2013 Win64'; $MSBUILDVER = "12.0"; }
+
+# clear any previous build and create & enter the build directory
+$buildDir = Join-Path "$projectRoot" "_build"
+Write-Output "Creating (or cleaning if already existing) the folder $buildDir for project files and outputs"
+Remove-Item -Path $buildDir -Recurse -Force -ErrorAction SilentlyContinue | Out-Null
+New-Item -ItemType Directory -Path $buildDir -ErrorAction SilentlyContinue | Out-Null
+Push-Location $buildDir
+
+# check cmake is installed
+if ($null -eq (Get-Command "cmake.exe" -ErrorAction SilentlyContinue)) 
+{ 
+    $installCmake = Read-Host "Unable to find cmake in your PATH - would you like to download and install automatically? [yes/no]"
+   
+    if ($installCmake -eq "y" -or $installCmake -eq "yes") {
+        # download cmake 3.17.3 64-bit and run MSI for user
+        $client = New-Object System.Net.WebClient        
+        $tempDownloadFile = New-TemporaryFile
+        $cmakeMsiFile = "$tempDownloadFile.cmake-3.17.3-win64-x64.msi"
+        Write-Output "Downloading cmake (temporary file location $cmakeMsiFile)"
+        Write-Output "Note: select the option to add cmake to path for this script to operate"
+        $client.DownloadFile("https://github.com/Kitware/CMake/releases/download/v3.17.3/cmake-3.17.3-win64-x64.msi", "$cmakeMsiFile")
+        Start-Process $cmakeMsiFile -Wait
+        Remove-Item $cmakeMsiFile
+        Write-Output "Cmake should have installed, this script will now exit because of path updates - please now re-run this script"
+        exit
+    }
+    else{
+        Write-Output "Quitting because cmake is required"     
+        exit
+    }
+}
+
+if ( $CXX11 -eq "OFF" ) {
+    # get pthreads (this is legacy, and is only availble in nuget for VS2015 and VS2013)
+    if ( $VS_VERSION -gt 2015 ) { 
+        Write-Output "Pthreads is not recommended for use beyond VS2015 and is not supported by this build script - aborting build"
+        exit
+    }
+    if ( $DEVENV_PLATFORM -eq 'Win32' ) { 
+        nuget install cinegy.pthreads-win32-$VS_VERSION -version 2.9.1.24 -OutputDirectory ../_packages
+    }
+    else {        
+        nuget install cinegy.pthreads-win64-$VS_VERSION -version 2.9.1.24 -OutputDirectory ../_packages
+    }
+}
+
+if ($STATIC_LINK_SSL -eq "ON") {
```

Comment:
I think before informing about enforced turning on encryption, you should check if it isn't on already.

Reply:
Yeah, that was really me just being lazy... I was in a bit of a hurry to get this finished before the weekend :)

Reply:
Then better let's finish this before the next Friday :D